prusti-viper/src/encoder/mir/procedures/encoder.rs

```diff
@@ -104,7 +104,8 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
             deallocate_returns,
             assert_postconditions,
         );
-        self.encode_body(&mut procedure_builder)?;
+        let rd_perm: u32 = self.lifetimes.lifetime_count();
```

Comment:
Why is this not a field on `ProcedureEncoder`, but an additional argument/parameter that needs to be threaded through?

Reply:
yes, done.